### PR TITLE
Cloudformation template for this stack

### PR DESCRIPTION
**Pull request by arunasank**
Ref: https://github.com/hotosm/OpenMapKitServer/issues/90

Setting up a CloudFormation template for OMK Server.

cc/ @hotosm/tech

### CODE REVIEW

File: cloudformation/OpenMapKitServer.template.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+          '   python-wheel \ ',
+          ' && apt-get clean \ ',
+          '&& rm -rf /var/lib/apt/lists/* ',
+          'npm install -g yarn \ ',
+          '&& rm -rf /root/.npm',
+          'mkdir -p /app',
+          'git clone https://github.com/hotosm/OpenMapKitServer.git && cp -R OpenMapKitServer/* /app/ && cd app',
+          'pip install -r requirements.txt',
+          'yarn && rm -rf /root/.cache/yarn',
+          'yarn pushbuild',
```
Comment (willemarcel) [inline cloudformation/OpenMapKitServer.template.js]: This command is to send the frontend build to git.
On the cfn, we need to pull the `dist` branch and put it on the `frontend/build/` folder

File: cloudformation/OpenMapKitServer.template.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        },
+        Cooldown: 300
+      }
+  },
+  OpenMapKitServerLaunchConfiguration: {
+    Type: 'AWS::AutoScaling::LaunchConfiguration',
+      Properties: {
+        IamInstanceProfile: cf.ref('OpenMapKitServerEC2InstanceProfile'),
+        ImageId: 'ami-0e4372c1860d7426c',
+        InstanceType: 'm3.large',
```
Comment (willemarcel) [inline cloudformation/OpenMapKitServer.template.js]: Currently we are using `t2.medium` to the Nepal instance.

File: cloudformation/OpenMapKitServer.template.js (hunk 1 above)
Comment (arunasank) [reply cloudformation/OpenMapKitServer.template.js]: 👋 @willemarcel Thanks for reviewing. ♥️ 

I am not sure I follow ^ entirely - is it something like `git checkout dist && cp . -R frontend/build && git checkout master`

File: cloudformation/OpenMapKitServer.template.js (hunk 2 above)
Comment (arunasank) [reply cloudformation/OpenMapKitServer.template.js]: Noted. 🙂

File: cloudformation/OpenMapKitServer.template.js (hunk 1 above)
Comment (willemarcel) [reply cloudformation/OpenMapKitServer.template.js]: My suggestion. Let's define a variable to the OpenMapKit version. I don't know where is more appropriate to set it on the cfn...

```
VERSION=v1.3.0
git clone https://github.com/hotosm/OpenMapKitServer.git /app/ && cd app && git checkout $VERSION
```

So we can download the frontend build and copy it to its proper directory with

```
wget https://github.com/hotosm/OpenMapKitServer/archive/${VERSION}-frontend.tar.gz -P /tmp/
rm frontend/build/* -R
tar -xvzf /tmp/${VERSION}-frontend.tar.gz -C frontend/build/ --strip 1
```

**Comment by arunasank** [comment]
gif

Hi folks! Thanks to help from @willemarcel, I managed to finish the first iteration of the infrastructure work for the OMK server. 💃 🎆 🎈 

## What has changed
* The entire infrastructure is set up on AWS using CloudFormation. This is how the architecture looks like currently:
screen shot 2019-02-13 at 10 06 32 pm

* We no longer have additional proxying from nginx, since Cloudfront and the ELB act as a 2 layer proxy
* Deploys to the stack, which happen from the console currently will happen via the command line.

## What remains to be done
https://github.com/hotosm/OpenMapKitServer/issues/93

Please jump on in if you're interested in any of the tasks in the above ticket!

**Comment by arunasank** [comment]
Feel free to test the newly set up instance on http://omk-test.hotosm.org/, and report any issues that you find. 🙂

**Comment by arunasank** [comment]
This happened! Merging this with master. : )

**Pull request merged**